File: Models/PropagationSetting.cs

using Umbraco.Community.MCPS.Models.Enums;

namespace Umbraco.Community.MCPS.Models;

public class PropagationSetting
{
    public int? Id { get; set; }
    // Contains the rules defining the setting itself
    public required string Name { get; set; }
    public List<PropagationRank> PropagationPriority { get; set; } = [PropagationRank.Relevant, PropagationRank.Newest, PropagationRank.MostPopular];
    public DuplicationStrategy DuplicationStrategy { get; set; } = DuplicationStrategy.UniquePage;
    public required List<string> ContentTypes { get; set; }
    public required string PropertyAlias { get; set; }
    public PropagationSetting? FallbackSetting { get; set; }
}

File: Client/src/models/propagationSetting.ts
import { PropagationRank, DuplicationStrategy } from './enums';

export class PropagationSetting {
    id?: number;
    name?: string;
    propagationPriority: PropagationRank[];
    duplicationStrategy: DuplicationStrategy;
    contentTypes?: string[];
    propertyAlias?: string;
    fallbackSetting?: PropagationSetting;

    constructor(params: {
        id?: number;
        name?: string;
        propagationPriority?: PropagationRank[];
        duplicationStrategy?: DuplicationStrategy;
        contentTypes?: string[];
        propertyAlias?: string;
        fallbackSetting?: PropagationSetting;
    }) {
        this.id = params.id;
        this.name = params.name;
        this.propagationPriority = params.propagationPriority ?? [
            PropagationRank.Relevant,
            PropagationRank.Newest,
            PropagationRank.MostPopular,
        ];
        this.duplicationStrategy = params.duplicationStrategy ?? DuplicationStrategy.UniquePage;
        this.contentTypes = params.contentTypes;
        this.propertyAlias = params.propertyAlias;
        this.fallbackSetting = params.fallbackSetting;
    }
}
